pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```
@@ -103,23 +103,26 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	log.Info("Maintaining Shoot")
 
 	var (
-		maintainedShoot = shoot.DeepCopy()
-		operations      []string
+		maintainedShoot             = shoot.DeepCopy()
+		operations                  []string
+		reasonForImageUpdatePerPool []string
 	)
 
 	cloudProfile := &gardencorev1beta1.CloudProfile{}
 	if err := r.Client.Get(ctx, kubernetesutils.Key(shoot.Spec.CloudProfileName), cloudProfile); err != nil {
 		return err
 	}
 
-	reasonForImageUpdatePerPool, err := maintainMachineImages(log, maintainedShoot, cloudProfile)
-	if err != nil {
-		// continue execution to allow the kubernetes version update
-		log.Error(err, "Failed to maintain Shoot machine images")
+	if !shoot.IsWorkerless() {
+		reasonForImageUpdatePerPool, err := maintainMachineImages(log, maintainedShoot, cloudProfile)
+		if err != nil {
+			// continue execution to allow the kubernetes version update
+			log.Error(err, "Failed to maintain Shoot machine images")
+		}
+		operations = append(operations, reasonForImageUpdatePerPool...)
 	}
-	operations = append(operations, reasonForImageUpdatePerPool...)
 
-	reasonForKubernetesUpdate, err := maintainKubernetesVersion(log, maintainedShoot.Spec.Kubernetes.Version, maintainedShoot.Spec.Maintenance.AutoUpdate.KubernetesVersion, cloudProfile, func(v string) error {
+	reasonForKubernetesUpdate, err := maintainKubernetesVersion(log, maintainedShoot.Spec.Kubernetes.Version, *maintainedShoot.Spec.Maintenance.AutoUpdate.KubernetesVersion, cloudProfile, func(v string) error {
```

its a bit strange that after calling `maintainKubernetesVersion` and ignoring the error that it returns it goes ahead and returns an error if the version is not a correct semantic version. The correct semantic version check should be done prior to calling `maintainKubernetesVersion`

---

Semantic version is being checked for `maintained shoot`, i.e. if the error would have occurred in the `maintainKubernetesVersion`  then the maintained shoot Kubernetes version would not have been updated and old shoot kubernetes version can't be non-semantic.